core/src/language/kotlin.rs

```diff
@@ -119,16 +120,50 @@ impl Language for Kotlin {
         self.write_comments(w, 0, &ty.comments)?;
         let type_name = format!("{}{}", &self.prefix, ty.id.original);
 
-        writeln!(
-            w,
-            "typealias {}{} = {}\n",
-            type_name,
-            (!ty.generic_types.is_empty())
-                .then(|| format!("<{}>", ty.generic_types.join(", ")))
-                .unwrap_or_default(),
-            self.format_type(&ty.r#type, ty.generic_types.as_slice())
-                .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?
-        )?;
+        if self.is_inline(&ty.decorators) {
+            writeln!(w, "@Serializable")?;
+            writeln!(w, "@JvmInline")?;
+            writeln!(w, "value class {}{}(", self.prefix, ty.id.renamed)?;
+
+            self.write_element(
```

Comment:
Is this block equivalent to a similar block in the `write_struct` code?

Reply:
The main difference between this and the one in `write_struct` is that structs are being translated into `data` [classes](https://kotlinlang.org/docs/data-classes.html) (or an `object` if it contains no data) and this is specifically a `value` [class](https://kotlinlang.org/docs/inline-classes.html).